tofhir-server/src/main/scala/io/tofhir/server/service/SchemaDefinitionService.scala

```diff
@@ -97,8 +99,72 @@ class SchemaDefinitionService(schemaRepository: ISchemaRepository, mappingReposi
    */
   def inferSchema(inferTask: InferTask): Future[Option[SchemaDefinition]] = {
     // Execute SQL and get the dataFrame
-    val dataFrame = SourceHandler.readSource(inferTask.name, SparkConfig.sparkSession,
-      inferTask.sourceContext, inferTask.sourceSettings.head._2, None, None, Some(1))
+    val dataFrame = try {
+      SourceHandler.readSource(inferTask.name, SparkConfig.sparkSession,
+        inferTask.sourceContext, inferTask.sourceSettings.head._2, None, None, Some(1))
+    } catch {
+      case e: Throwable =>
+        println(e.getClass)
+        val errorClass = e.getClass
+        val errorMessage: String = e.getMessage
+
+        // This error is thrown if the extension of the file is not expected
+        if(errorClass == classOf[scala.NotImplementedError]){
+          val filePath = inferTask.sourceContext.asInstanceOf[FileSystemSource].path;
+          throw BadRequest("Unsupported file", s" The file ${filePath} is not supported. Correct the extension of the file.")
+        }
+
+        // Gives the error if path of the source file is wrong
+        else if (errorClass == classOf[org.apache.spark.sql.AnalysisException]) {
+          val dataFolderPath: String = inferTask.sourceSettings("source").asInstanceOf[FileSystemSourceSettings].dataFolderPath
+          throw ResourceNotFound("Path not found", s"The file to be inferred cannot be found in the path: ${dataFolderPath}")
+        }
+
+        // Syntax errors in preprocess SQL field in front-end
+        else if (errorClass == classOf[org.apache.spark.sql.catalyst.parser.ParseException]) {
+          // Capitalize and replace("\n", " ") is needed to read the message in the front-end properly
+          throw InternalError("Preprocess SQL syntax error", errorMessage.capitalize.replace("\n", " "))
+        }
+
+        /**
+         * If the error is an instance of SQL error handle it by its SQLState
+         * Reference: https://www.ibm.com/docs/en/i/7.4?topic=codes-listing-sqlstate-values
+         */
+        else if(classOf[java.sql.SQLException].isAssignableFrom(errorClass)){
+          val SQLError: SQLException = e.asInstanceOf[SQLException]
```

Comment:
rename this and similar fields so that they do not start with capital abbreviations e.g. sqlError